In [24]:
# Nama    : Diah Anggraini
# NPM     : 24083010051
# Mata Kuliah : Matematika Diskrit A

In [11]:
import pandas as pd
import random
import math

In [12]:
n_values_1 = sorted(random.sample(range(0, 1001), 10))

rows1 = []
for n in n_values_1:
    sisi_kiri = sum(2*k-1 for k in range(1, n+1))
    sisi_kanan = n**2
    pernyataan = (sisi_kiri == sisi_kanan)
    rows1.append([n, sisi_kiri, sisi_kanan, pernyataan])

df1 = pd.DataFrame(rows1, columns=["n","Sisi_Kiri","Sisi_Kanan","Pernyataan_Benar?"])

print("=== No.1: 1+3+5+...(2n-1)=n^2 ===")
print(df1)

=== No.1: 1+3+5+...(2n-1)=n^2 ===
     n  Sisi_Kiri  Sisi_Kanan  Pernyataan_Benar?
0   23        529         529               True
1   46       2116        2116               True
2   62       3844        3844               True
3  346     119716      119716               True
4  454     206116      206116               True
5  497     247009      247009               True
6  499     249001      249001               True
7  615     378225      378225               True
8  702     492804      492804               True
9  881     776161      776161               True


Penjelasan :
Output tersebut menunjukkan bahwa untuk sepuluh nilai acak n mulai dari 23 hingga 881, nilai sisi kiri berupa penjumlahan bilangan ganjil pertama hingga suku ke-n selalu sama dengan sisi kanan yaitu n^2. Baik pada nilai n yang relatif kecil (23, 46, 62) maupun pada nilai yang besar (346, 454, 615, 702, 881), hasilnya konsisten menunjukkan kesetaraan. Sehingga kesimpulan yang bisa diambil adalah rumus 1+3+5+…+(2n−1)=n^2 benar untuk semua nilai n yang diuji, dan mendukung bahwa rumus ini berlaku umum untuk n>=0.

In [27]:
n_values_5 = sorted(random.sample(range(-10, 101), 10))

rows5 = []
for n in n_values_5:
    sisi_kiri = 2**n
    sisi_kanan = n**2
    pernyataan = (sisi_kiri > sisi_kanan)
    rows5.append([n, sisi_kiri, sisi_kanan, pernyataan])

df5 = pd.DataFrame(rows5, columns=["n","Sisi_Kiri (2^n)","Sisi_Kanan (n^2)","Pernyataan_Benar?"])

print("\n=== No.5: 2^n > n^2 ===")
print(df5)


=== No.5: 2^n > n^2 ===
    n                 Sisi_Kiri (2^n)  Sisi_Kanan (n^2)  Pernyataan_Benar?
0  -5                         0.03125                25              False
1   0                               1                 0               True
2   6                              64                36               True
3  33                      8589934592              1089               True
4  36                     68719476736              1296               True
5  40                   1099511627776              1600               True
6  80       1208925819614629174706176              6400               True
7  81       2417851639229258349412352              6561               True
8  91    2475880078570760549798248448              8281               True
9  99  633825300114114700748351602688              9801               True


Penjelasan :
Output tersebut menunjukkan bahwa untuk n negatif seperti –5, nilai 2^n jauh lebih kecil dibandingkan n^2 sehingga pernyataannya salah. Sebaliknya, mulai dari n = 0 dan pada nilai-nilai positif yang cukup besar seperti 6, 33, 36, 40, 80, 81, 91, dan 99, nilai 2^n jauh melampaui n^2 sehingga pernyataan 2^n > n^2 selalu benar. Sehingga kesimpulan yang bisa diambil adalah pernyataan ini salah untuk n negatif tetapi benar untuk n = 0 dan seluruh nilai n positif yang besar.